src/search.rs

```diff
@@ -144,14 +144,10 @@ pub struct Query<'a> {
     /// Default: `20`
     #[serde(skip_serializing_if = "Option::is_none")]
     pub limit: Option<usize>,
-    /// Filters applied to documents.
+    /// Filter applied to documents.
     /// Read the [dedicated guide](https://docs.meilisearch.com/reference/features/filtering.html) to learn the syntax.
     #[serde(skip_serializing_if = "Option::is_none")]
-    pub filters: Option<&'a str>,
-    /// Facet names and values to filter on.
-    /// Read [this page](https://docs.meilisearch.com/reference/features/search_parameters.html#facet-filters) for a complete explanation.
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub facet_filters: Option<&'a [&'a [&'a str]]>,
+    pub filter: Option<&'a str>,
```

Comment:
The type of fillter is the same as `facet_filters` but can also accept a `str` as value. So it's a mix of both `filter` and `facet_filters`. 
The following syntaxes are possible: 

nested array:
```json
["something > 1", "genre=comedy", ["genre=horror", "title=batman"]]
```
only string: 
```json
"something > 1 AND genre=comedy AND (genre=horror OR title=batman)"
```

of course both are also mixable
```json
["something > 1 AND genre=comedy", "genre=horror OR title=batman"]
```

Reply:
After some thought we may also just only accept the string format. What do you think?